src/main/java/codeu/model/data/Message.java

```diff
@@ -34,13 +36,19 @@
    * @param author the ID of the User who sent this Message
    * @param content the text content of this Message
    * @param creation the creation time of this Message
+   * @param parseImages whether or not links in the message should be found and
+   *          added as loaded pictures at end. Based on whether new message or loaded.
    */
-  public Message(UUID id, UUID conversation, UUID author, String content, Instant creation) {
+  public Message(UUID id, UUID conversation, UUID author, String content, Instant creation, 
+                 boolean parseImages) {
     this.id = id;
     this.conversation = conversation;
     this.author = author;
-    this.content = content;
     this.creation = creation;
+    if(parseImages) 
+        this.content = parseImages(content);
+    else
+        this.content = content;
```

Comment:
Instead of overloading the content field with the image link.  I'd add a new field to Message which is a List<Image>.  The Image class should contain a "url" field.  We need to start abstracting the concept of an Image.

Roughly, the relationship would be a many to many relationship between Message and Image.  Each Image has a "url" field, and another field for description if we manage to get the vision API working.  Image should be a model and we should have it hooked up to the persistence layer.  Image should also have a "getHtml()" function which turns image.url into valid html like in your function below.